bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDiscoveryService.java

```diff
@@ -0,0 +1,437 @@
+                    packet = null;
+                } catch (SocketTimeoutException e) {
+                    // socket.receive() will time out every 1 second so the thread won't block
+                }
+            }
+        } catch (IOException e) {
+            if (!closing) {
+                logger.warn("listenMulticast() error '{}'", e.getMessage());
+            }
+        } finally {
+            if (socket != null && networkInterface != null) {
+                try {
+                    socket.leaveGroup(SDDP_GROUP, networkInterface);
+                } catch (IOException e) {
+                    if (!closing) {
+                        logger.warn("listenMulticast() error '{}'", e.getMessage());
+                    }
+                }
+                socket.close();
+            }
+        }
+    }
+
+    /**
+     * Send a single outgoing SEARCH 'ping' and then continue to listen for incoming SDDP unicast responses until the
+     * loop time elapses or the thread is externally interrupted.
+     */
+    private void listenActiveScanUnicast() {
+        // get a free port number
+        int port;
+        try (ServerSocket portFinder = new ServerSocket(0)) {
+            port = portFinder.getLocalPort();
+        } catch (IOException e) {
+            logger.warn("listenUnicast() port finder error '{}'", e.getMessage());
+            return;
+        }
+
+        try (DatagramSocket socket = new DatagramSocket(port)) {
+            String ipAddress = networkAddressService.getPrimaryIpv4HostAddress();
+            NetworkInterface networkInterface = NetworkInterface.getByInetAddress(InetAddress.getByName(ipAddress));
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("listenUnicast() starting on '{}:{}' on interface '{}'", ipAddress, port,
+                        networkInterface.getDisplayName());
+            }
+
+            socket.setOption(StandardSocketOptions.IP_MULTICAST_IF, networkInterface);
+            socket.setSoTimeout((int) SOCKET_TIMOUT.toMillis());
+
+            DatagramPacket packet;
+            byte[] buffer;
+
+            // send search request
+            String search = String.format(SEARCH_REQUEST_BODY_FORMAT, ipAddress, port);
+            buffer = search.getBytes(StandardCharsets.UTF_8);
+            packet = new DatagramPacket(buffer, buffer.length, new InetSocketAddress(SDDP_IP_ADDRESS, SDDP_PORT));
+            socket.send(packet);
+            packet = null;
+            logger.debug("Packet sent to '{}:{}' content:\r\n{}", SDDP_IP_ADDRESS, SDDP_PORT, search);
+
+            final Instant listenDoneTime = Instant.now().plus(SEARCH_LISTEN_DURATION);
+
+            // loop listen for responses
+            while (Instant.now().isBefore(listenDoneTime) && !Thread.currentThread().isInterrupted()) {
+                try {
+                    if (packet == null) {
+                        packet = new DatagramPacket(buffer, buffer.length);
```

Comment:
Re-using the send buffer was causing the received packet content to be truncated:
```suggestion
            final Instant listenDoneTime = Instant.now().plus(SEARCH_LISTEN_DURATION);
            byte[] readBuffer = new byte[READ_BUFFER_SIZE];

            // loop listen for responses
            while (Instant.now().isBefore(listenDoneTime) && !Thread.currentThread().isInterrupted()) {
                try {
                    if (packet == null) {
                        packet = new DatagramPacket(readBuffer, readBuffer.length);
```

Reply:
Ok